plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -662,34 +666,30 @@ def _get_repo_unit_nevra(self, repo_id):
             repo_unit_nevra.add(models.NEVRA(*scalar))
         return repo_unit_nevra
 
-    def _get_pkglist_to_publish(self, erratum_unit):
+    def _get_repo_module_nsvca(self, repo_id):
         """
-        Make a list of packages which will be listed in the published erratum.
-
-        Only packages contained in the repository being published will be included in this
-        errata's pkglist.
-
-        This merges multiple Pulp pkglists into a single pkglist for publication.
-
-        :param erratum_unit: the erratum unit to analyze
-        :param type: pulp_rpm.plugins.db.models.Errata
-        :return: pkglist to publish
-        :rtype: dict
+        Return a set of NVSCA module tuples, associated with the repo_id, that the errata being
+        processed might affect
         """
-        repo_id = self.get_repo().id
+        querysets = repo_controller.get_unit_model_querysets(repo_id, models.Modulemd)
+        return set(
+            itertools.chain(*[q.scalar(*models.Modulemd.unit_key_fields) for q in querysets]))
 
-        new_pkglist = {
+    def _new_pkglist(self, repo_id, index):
+        # start a new pkglist/collection (for a module), ammend the pkglist name, record
+        # the module and add the pkglist into the returned list
+        proto_pkglist = {
             # Due to a quirk in how yum merges pkglists in errata that appear in multiple
             # repos, the pkglist collection name needs to be unique per-repo when published.
             # repo_id is already unique and presumably validated, so use that. This is normally a
             # human-readable string, like "A Human-Readable Repository Description". Pulp doesn't
             # require storing that sort of string with a repo, so repo_id is the best we've got.
-            'name': repo_id,
+            'name': '{}_{}'.format(repo_id, index),
```

Comment:
repo_id + index as a name looks good to me. I think it's good enough and including a module name might not be necessary. At the same time it can be helpful for testing/troubleshooting purposes to have a module name as well, not essential though.

Reply:
OK, would templates like`foo_repo_007_default` vs `bar_repo_42_betelgeuse` fit the non-modular vs modular collection naming respectively?